jgrapht-demo/src/main/java/org/jgrapht/demo/GraphMLDemo.java

```diff
@@ -0,0 +1,326 @@
+            new EdgeProvider<CustomVertex, DefaultWeightedEdge>()
+            {
+                @Override
+                public DefaultWeightedEdge buildEdge(
+                    CustomVertex from, CustomVertex to, String label,
+                    Map<String, String> attributes)
+                {
+                    return new DefaultWeightedEdge();
+                }
+            };
+
+        // create GraphML importer
+        GraphMLImporter<CustomVertex, DefaultWeightedEdge> importer =
+            new GraphMLImporter<>(vertexProvider, edgeProvider);
+
+        return importer;
+    }
+
+    /**
+     * Main demo method
+     * 
+     * @param args command line arguments
+     */
+    public static void main(String[] args)
+    {
+        /*
+         * Generate complete graph.
+         * 
+         * Vertices have random colors and edges have random edge weights.
+         */
+        DirectedPseudograph<CustomVertex, DefaultWeightedEdge> graph1 =
+            new DirectedWeightedPseudograph<>(DefaultWeightedEdge.class);
+        CompleteGraphGenerator<CustomVertex, DefaultWeightedEdge> completeGenerator =
+            new CompleteGraphGenerator<>(SIZE);
+        VertexFactory<CustomVertex> vFactory = new VertexFactory<CustomVertex>()
+        {
+            private int id = 0;
+
+            @Override
+            public CustomVertex createVertex()
+            {
+                CustomVertex v = new CustomVertex(String.valueOf(id++));
+                if (generator.nextBoolean()) {
+                    v.setColor(Color.BLACK);
+                } else {
+                    v.setColor(Color.WHITE);
+                }
+                return v;
+            }
+
+        };
+        System.out.println("-- Generating complete graph");
+        completeGenerator.generateGraph(graph1, vFactory, null);
+
+        // assign random weights
+        for (DefaultWeightedEdge e : graph1.edgeSet()) {
+            graph1.setEdgeWeight(e, generator.nextInt(100));
+        }
+
+        // now export and import back again
+        try {
+            // export as string
+            System.out.println("-- Exporting graph as GraphML");
+            GraphExporter<CustomVertex, DefaultWeightedEdge> exporter = createExporter();
+            Writer writer = new StringWriter();
+            exporter.exportGraph(graph1, writer);
+            String graph1AsGraphML = writer.toString();
+
+            // display
+            System.out.println(graph1AsGraphML);
+
+            // import it back
+            System.out.println("-- Importing graph back from GraphML");
+            Graph<CustomVertex, DefaultWeightedEdge> graph2 =
+                new DirectedWeightedPseudograph<>(DefaultWeightedEdge.class);
+            GraphImporter<CustomVertex, DefaultWeightedEdge> importer = createImporter();
+            importer.importGraph(graph2, new StringReader(graph1AsGraphML));
+            
+            // export again as string
```

Comment:
What is the purpose of exporting it again, ie.. what is the learning value here?